open-machine-learning-jupyter-book/ml-fundamentals/Build a Web App to use a Mchine Learning model.md

```diff
@@ -0,0 +1,352 @@
+
+```output
+web-app/
+    static/
+    css/
+    templates/
+notebook.ipynb
+ufo-model.pkl
+```
+
+✅ Refer to the solution folder for a view of the finished app
+
+The first file to create in _web-app_ folder is **requirements.txt** file. Like _package.json_ in a JavaScript app, this file lists dependencies required by the app. In **requirements.txt** add the lines:
+
+```text
+scikit-learn
+pandas
+numpy
+flask
+```
+
+Now, run this file by navigating to _web-app_:
+
+```output
+cd web-app
+```
+
+In your terminal type `pip install`, to install the libraries listed in _requirements.txt_:
+
+```output
+pip install -r requirements.txt
+```
+
+Now, you're ready to create three more files to finish the app:
+
+    1. Create **app.py** in the root.
+    2. Create **index.html** in _templates_ directory.
+    3. Create **styles.css** in _static/css_ directory.
+
+Build out the _styles.css_ file with a few styles:
+
+```css
+body {
+    width: 100%;
+    height: 100%;
+    font-family: 'Helvetica';
+    background: black;
+    color: #fff;
+    text-align: center;
+    letter-spacing: 1.4px;
+    font-size: 30px;
+}
+
+input {
+    min-width: 150px;
+}
+
+.grid {
+    width: 300px;
+    border: 1px solid #2d2d2d;
+    display: grid;
+    justify-content: center;
+    margin: 20px auto;
+}
+
+.box {
+    color: #fff;
+    background: #2d2d2d;
+    padding: 12px;
+    display: inline-block;
+}
+```
+
+Next, build out the _index.html_ file:
+
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+    <meta charset="UTF-8">
+    <title>🛸 UFO Appearance Prediction! 👽</title>
+    <link rel="stylesheet" href="{{ url_for('static', filename='css/styles.css') }}">
+    </head>
+
+    <body>
+    <div class="grid">
+
+        <div class="box">
+
+        <p>According to the number of seconds, latitude and longitude, which country is likely to have reported seeing a UFO?</p>
+
+        <form action="{{ url_for('predict')}}" method="post">
+            <input type="number" name="seconds" placeholder="Seconds" required="required" min="0" max="60" />
+            <input type="text" name="latitude" placeholder="Latitude" required="required" />
+            <input type="text" name="longitude" placeholder="Longitude" required="required" />
+            <button type="submit" class="btn">Predict country where the UFO is seen</button>
+        </form>
+
+        <p>{{ prediction_text }}</p>
+
+        </div>
+
+    </div>
+
+    </body>
+</html>
+```
+
+```{note}
+Take a look at the templating in this file. Notice the 'mustache' syntax around variables that will be provided by the app, like the prediction text: `{{}}`. There's also a form that posts a prediction to the `/predict` route.
+
+Finally, you're ready to build the python file that drives the consumption of the model and the display of predictions:
+```
+
+In `app.py` add:
+
+```python
```

Comment:
Correct! If we want the code to be executed while building the book, we use {code-cell}. Otherwise, we use python code block directly.